packages/runtime/container-runtime/src/test/blobManager.stashed.spec.ts

```diff
@@ -71,6 +86,99 @@ describe("BlobManager.stashed", () => {
 	});
 
 	it("Stashed blob", async () => {
+		const createResponse = new Deferred<ICreateBlobResponse>();
+		const blobManager = createBlobManager({
+			sendBlobAttachOp(_localId, _storageId) {},
+			stashedBlobs: {},
+			getStorage: () =>
+				failProxy<IDocumentStorageService>({
+					createBlob: async () => {
+						return createResponse.promise;
+					},
+				}),
+		});
+		const blob: ArrayBufferLike = stringToBuffer("content", "utf8");
+		const sameBlobAsStashedP = blobManager.createBlob(blob);
+		const pendingBlobsP = blobManager.attachAndGetPendingBlobs();
+		const sameBlobAsStashed = await sameBlobAsStashedP;
+		sameBlobAsStashed.attachGraph();
+		const pendingBlobs = await pendingBlobsP;
+		const blobManager2 = createBlobManager({
+			stashedBlobs: pendingBlobs,
+			getStorage: () =>
+				failProxy<IDocumentStorageService>({
+					createBlob: async () => {
+						return createResponse.promise;
+					},
+				}),
+		});
+		assert.strictEqual(blobManager2.hasPendingStashedUploads(), true);
+	});
+
+	it("Repro bug: Process blob and complete stashed upload after", async () => {
+		const createResponse = new Deferred<ICreateBlobResponse>();
+		const blobManager = createBlobManager({
+			sendBlobAttachOp(_localId, _storageId) {},
+			stashedBlobs: {},
+			getStorage: () =>
+				failProxy<IDocumentStorageService>({
+					createBlob: async () => {
+						return createResponse.promise;
+					},
+				}),
+			localBlobIdGenerator: () => "stubbed-local-id",
+		});
+
+		const blob: ArrayBufferLike = stringToBuffer("content", "utf8");
+		const sameBlobAsStashedP = blobManager.createBlob(blob);
+		const pendingBlobsP = blobManager.attachAndGetPendingBlobs();
+		const sameBlobAsStashed = await sameBlobAsStashedP;
+		sameBlobAsStashed.attachGraph();
+		const pendingBlobs = await pendingBlobsP;
+
+		const createResponse2 = new Deferred<ICreateBlobResponse>();
+		const blobManager2 = createBlobManager({
+			stashedBlobs: pendingBlobs,
+			getStorage: () =>
+				failProxy<IDocumentStorageService>({
+					createBlob: async () => {
+						return createResponse2.promise;
+					},
+				}),
+		});
+		assert.strictEqual(blobManager2.hasPendingStashedUploads(), true);
+
+		blobManager2.processBlobAttachMessage(
+			{
+				clientId: "client-id",
+				minimumSequenceNumber: 1,
+				referenceSequenceNumber: 1,
+				sequenceNumber: 1,
+				type: "blobAttach",
+				metadata: {
+					localId: "stubbed-local-id",
+					blobId: "stubbed-storage-id",
+				},
+				timestamp: Date.now(),
+			},
+			true,
+		);
+		await Promise.race([
+			new Promise<void>((resolve) => setTimeout(() => resolve(), 10)),
+			blobManager2.trackPendingStashedUploads(),
+		]);
+		createResponse2.resolve({
+			id: "new-storage-id",
+		});
+		try {
+			await blobManager2.waitForStashedBlobs();
```

Comment:
i thought i left this comment already, but i don't see it. Why does this only need to be done in test? doesn't anything else need to take this into account?